src/state/nftMarket/hooks.ts

```diff
@@ -99,6 +103,35 @@ export const useGetNFTInitializationState = () => {
   return useSelector((state: State) => state.nftMarket.initializationState)
 }
 
+export const useApprovalNfts = (nftsInWallet: NftToken[]) => {
+  const nftApprovalCalls = useMemo(
+    () =>
+      nftsInWallet.map((nft: NftToken) => {
+        const { tokenId, collectionAddress } = nft
+
+        return {
+          address: collectionAddress,
+          name: 'getApproved',
+          params: [tokenId],
+        }
+      }),
+    [nftsInWallet],
+  )
+
+  const { data } = useSWRMulticall(erc721Abi, nftApprovalCalls)
```

Comment:
What is the advantage of using swr here?

Reply:
Will fetch by selectedNft affect user experience when switching selected? Just wondering if it is good enough to not use multicall here

Reply:
Without multicall, whenever user selects a pic, a loading status is shown, which is not a good UX. For this feature, it's reasonable to call it in one batch.